test/int/adapter.test.ts

```diff
@@ -85,5 +89,52 @@ suite('Chrome Debug Adapter etc', function () {
                 dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
             ]);
         });
+
+        test('Should attach to existing instance of chrome and break on debugger statement', async () => {
+            const testProjectRoot = path.join(DATA_ROOT, 'intervalDebugger');
+            const breakFile = path.join(testProjectRoot, 'src/app.ts');
+            const DEBUGGER_LINE = 2;
+
+            server = createServer({ root: testProjectRoot });
+            server.listen(7890);
+            const browser = await puppeteer.launch(<puppeteer.LaunchOptions>{ headless: false, args: ['http://localhost:7890', '--remote-debugging-port=7777'] });
+
+            await Promise.all([
+                dc.configurationSequence(),
+                dc.initializeRequest().then(_ => {
+                    return dc.attachRequest(<IAttachRequestArgs>{ url: 'http://localhost:7890', port: 7777, webRoot: testProjectRoot });
+                }),
+                dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
+            ]);
+
+            await browser.close();
+        });
+
+        test('Should throw error when launching if chrome debug port is in use', async () => {
+
+            const testProjectRoot = path.join(DATA_ROOT, 'intervalDebugger');
+
+            server = createServer({ root: testProjectRoot });
+            server.listen(7890);
+
+            // browser already launched to the default port, and navigated away from about:blank
+            const browser = await puppeteer.launch(<puppeteer.LaunchOptions>{ headless: false, args: ['http://localhost:7890', '--remote-debugging-port=9222'] });
+            try {
+                await Promise.all([
+                    dc.configurationSequence(),
+                    dc.launch({ url: 'http://localhost:7890', timeout: 2000, webRoot: testProjectRoot }),
+                ]);
+            } catch (err) {
+                expect(err.message).to.satisfy( (x: string) => x.startsWith('Cannot connect to runtime process, timeout after 2000 ms'));
+            }
+
+            await browser.close();
+
+            console.log('killing all instances of chrome');
+            // force kill chrome here, as it will be left open by the debug adapter (same behavior as v1)
+            const killCmd = (process.platform === 'win32') ? 'taskkill /F /IM chrome.exe /T' : 'killall chrome';
```

Comment:
This code is similar to the code in testSetup.tearDown. Would it be possible to unify it?

Is this behavior undesirable? If so, we should add a TODO to fix it after we release v2.

Reply:
It can make troubleshooting easier. If you start Chrome, fail to attach, then kill Chrome, sometimes people might think that Chrome never actually started.

Reply:
I definitely think we need to look at changing the behavior for V2 after we get the initial release out, because what essentially happens here is that the debugger attaches properly but expects to find an "about:blank" page, which we launch to when we launch chrome. If a user already has an instance of chrome running with the debug flag, it won't have that about:blank page, and the debugger bails, even if it might have been able to recover.

For now I'll create a "kill all instance of chrome" function that will get rid of the duplicated code.